R/keyword_search.r

```diff
@@ -76,26 +78,46 @@ keyword_search <- function(x, keyword, path = FALSE, split_pdf = FALSE,
                            remove_hyphen = TRUE, token_results = TRUE,
                            heading_search = FALSE, heading_args = NULL,
                            convert_sentence = TRUE, 
-                           remove_equations = TRUE,
-                           split_pattern = "\\p{WHITE_SPACE}{3,}", ...) {
+                           remove_equations = FALSE,
+                           split_pattern = "\\p{WHITE_SPACE}{3,}",
+                           ocr_pdf_fun = function(...) {return("")},
+                           ocr_pdf_fun_params = nil,
+                           translate_fun = function(x,...) {return(x)},
+                           translate_fun_params = nil,
+                           translation_target_language="eng",
+                           ...) {
+  print(x)
   if(path) {
-    x <- pdftools::pdf_text(x)
+    path <- x
+    x  <- suppressMessages(pdftools::pdf_text(x))
+    output_dir_name <- paste0(dirname(path),"/outputs/")
+    dir.create(output_dir_name,showWarnings = F)
+    writeLines(x,file(paste0(dirname(path),"/outputs/",basename(path),".txt")) )

```

Comment:
@behrica I'm a bit wary of this piece of code to create the directory and write the lines to that directory without the user directly specifying the directory or opting in to saving the results. I saw similar code earlier on too, is this piece needed for the OCR?

Reply:
The reason for this is traceability.
My users like the fact, that the magic of the OCR + translation becomes visible in the form of files,
so they can archive the whole folder of original PDFs + OCR result + translation result.

This can eventually later explain, why certain things have not been found., which is important in our context.

Maybe there is a better way to do this in R.

Reply:
we could have an option to enable or disable this.

Same is true for progress logging.

As the eventually large pdfs gets uploaded to Azure, the "search" on lots of files could take hours.
(I implemented some caching, so a second search will not do it anymore),

Do you have a suggestion for this ?